File: python/1738_Find_Kth_Largest_XOR_Coordinate_Value.py
class Solution:
    def kthLargestValue(self, matrix: List[List[int]], k: int) -> int:
        row = len(matrix)
        col = len(matrix[0])
        maxK = row * col
        if k > maxK:
            return 0
        
        xorlist = [[0 for i in range(col)] for _ in range(row)]
        anslist = []
        for i in range(row):
            for j in range(col):
                if i == 0 and j == 0:
                    xorlist[i][j] = matrix[i][j]
                    anslist.append(xorlist[i][j])
                elif i == 0:
                    xorlist[i][j] = matrix[i][j] ^ xorlist[i][j-1]
                    anslist.append(xorlist[i][j])
                elif j == 0:
                    xorlist[i][j] = matrix[i][j] ^ xorlist[i-1][j]
                    anslist.append(xorlist[i][j])
                else:
                    xorlist[i][j] = matrix[i][j] ^ xorlist[i-1][j] ^ xorlist[i][j-1] ^ xorlist[i-1][j-1]
                    anslist.append(xorlist[i][j])
        
        anslist.sort()
        return anslist[-1*k]

File: java/1738_Find_Kth_Largest_XOR_Coordinate_Value.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

class Solution {
    public int kthLargestValue(int[][] matrix, int k) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[][] pre = new int[m+1][n+1];
        List<Integer> results = new ArrayList<Integer>();
        for(int i = 1 ; i <= m ; i++){
            for(int j = 1 ; j <= n ; j++){
                pre[i][j] = matrix[i-1][j-1] ^ pre[i-1][j] ^ pre[i][j-1] ^ pre[i-1][j-1];
                results.add(pre[i][j]);
            }
        }
        Collections.sort(results,
        new Comparator<Integer>(){
            public int compare(Integer num1, Integer num2){
                return num2 - num1;
            }
        });
        return results.get(k-1);
    }
}